SmartCarApplication/app/src/main/java/com/example/smartcarapplication/MainActivity.java

```diff
@@ -151,6 +151,10 @@ public void messageArrived(String topic, MqttMessage message) throws Exception {
                         }
                         bm.setPixels(colors, 0, IMAGE_WIDTH, 0, 0, IMAGE_WIDTH, IMAGE_HEIGHT);
                         mCameraView.setImageBitmap(bm);
```

Comment:
This would give an error because of where you removed the commented out ```mCameraView = findViewById(R.id.ImageView)```. It's annoying when Android Studio can't see such errors, but surprised IntelliJ didn't catch them @arrestoteles

Reply:
> > @Druner @um33 @adityak714 @Gabri1998 mind taking a look with a fresh pair of eyes and see if you can find out how to make this feature work? Thanks in advance!
> 
> @arrestoteles
> 
> I looked at the function messageArrived( ) (line 140), where in your conditional block,
> 
> `if (topic = smartcar/camera) -> ...` `else if (topic = smartcar/speed) -> ...`
> 
> seemed to be an issue maybe.
> 
> The MQTT connection on the Android end is handling multiple types of messages (camfeed messages, throttle messages and speed value messages), so it should be able to do that simultaneously, and not exclude the other if one topic's message is processed.
> 
> Currently, if the MQTT message is coming from the camera, it will disregard and not do anything if a message came from the smartcar/speed topic. So maybe change the else if (line 155) to an if only?

Thank you for the review and some good advice! ❤️ 

However, the if condition wasn't the problem. ❌ 

@sh4r10 helped me out with the bug and fixed the code. Apparently changing "speed_log" to "speedlog" did the trick. We assume that underscores are not allowed in id's in the activity_main.xml file. ✅

Reply:
Since its still a member variable to your class the compiler won't see any issue with it as java objects can implicitly be null... look for java nullability for ways to enforce certain invariants